### PR TITLE
Add data preservation

### PR DESCRIPTION
**Pull request by manq2010**
OOP school library: preserve data
---

In this PR, the following was done:

- [x] Saved all data into separate files
- [x] Preserved data is loaded on app restart
- [x] Handled missing file / folder

### CODE REVIEW

File: user_interface/app.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = BookUserInterface.new
+    @people = PersonUserInterface.new
+    @rentals = RentalUserInterface.new(@books, @people)
+  end
+
```
Comment (ShadyShawkat) [inline user_interface/app.rb:18]: - [ ] Please make sure that the data is preserved correctly. Currently, the data is preserved and saved while the app is opened. When the app is closed and reopened, you lose the previous session's data. As per [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_preserve_data.md#:~:text=On%20startup%20of%20your%20app%20load%20all%20data%20from%20files.) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: user_interface/app.rb (hunk 1 above)
Comment (manq2010) [reply user_interface/app.rb:18]: @ShadyShawkat please note that the required changes have been implemented as per the requirement

**Comment by AshakaE** [comment]
Hi @manq2010 , 

### Highlights ✨ 

- Pull request 👌 
- Git flow ✅ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**